lib/fastly_nsq/message_queue/listener.rb

```diff
@@ -13,7 +13,9 @@ def go
 
     def process_next_message
       message = consumer.pop
-      MessageProcessor.new(message).go
+      MessageProcessor.new(message.body).go
+      message.finish
+      consumer.terminate
```

Comment:
`INT` and `TERM` need to be trapped in the `go` loop above.

This allow us to `ensure` the closure of the `termination` of the consumer.